auth/src/main/java/com/firebase/ui/auth/data/client/CountryListLoadTask.java

```diff
@@ -38,267 +43,74 @@
         void onLoadComplete(List<CountryInfo> result);
     }
 
-    private static final int MAX_COUNTRIES = 291;
-
     private final Listener mListener;
 
-    public CountryListLoadTask(Listener listener) {
+    Map<String, Integer> countryInfoMap;
+    List<String> whitelistedCountryIsos;
+    List<String> blacklistedCountryIsos;
+
+    public CountryListLoadTask(Listener listener,
+                               List<String> whitelistedCountries,
+                               List<String> blacklistedCountries) {
         mListener = listener;
+        countryInfoMap = PhoneNumberUtils.getImmutableCountryIsoMap();
```

Comment:
@lsirac would it mess you up if this work was moved to `doInBackground`? I think originally there was a concern that the construction of this map was expensive enough to be done in the background (motivates the whole existence of this class).

If you can move it to there, then I am confident this is a safe change.

Reply:
Looking around, I think that just means you will want to lazy-initialize the big static members of `PhoneNumberUtils` and expose getters for them.

Reply:
Done.